{
  "comments": [
    {
      "key": {
        "uuid": "8710e4e8_33536dd0",
        "filename": "src/PSUSensor.cpp",
        "patchSetId": 5
      },
      "lineNbr": 181,
      "author": {
        "id": 1000521
      },
      "writtenOn": "2021-08-05T23:43:18Z",
      "side": 1,
      "message": "doesn\u0027t this end up incrementing error twice? \nsince fd \u003c 0 I expect line 192 \"read(fd, buffer.data()...)\" to return -1 and errno \u003d\u003d EBADF.\nThen line 223 incrementsError again.",
      "range": {
        "startLine": 181,
        "startChar": 12,
        "endLine": 181,
        "endChar": 26
      },
      "revId": "0588ee5a0357c3372903f06abcdf181a0404d20d",
      "serverId": "adbd0a64-1f21-4d83-b585-671fe73cb6e4",
      "unresolved": true
    },
    {
      "key": {
        "uuid": "d99ca13c_792ac3a7",
        "filename": "src/PSUSensor.cpp",
        "patchSetId": 5
      },
      "lineNbr": 181,
      "author": {
        "id": 1001162
      },
      "writtenOn": "2021-08-05T23:53:17Z",
      "side": 1,
      "message": "Right good catch, fixed.",
      "parentUuid": "8710e4e8_33536dd0",
      "range": {
        "startLine": 181,
        "startChar": 12,
        "endLine": 181,
        "endChar": 26
      },
      "revId": "0588ee5a0357c3372903f06abcdf181a0404d20d",
      "serverId": "adbd0a64-1f21-4d83-b585-671fe73cb6e4",
      "unresolved": false
    },
    {
      "key": {
        "uuid": "96b867ac_3206d62e",
        "filename": "src/PSUSensor.cpp",
        "patchSetId": 5
      },
      "lineNbr": 211,
      "author": {
        "id": 1000521
      },
      "writtenOn": "2021-08-05T23:52:39Z",
      "side": 1,
      "message": "is this correct? I don\u0027t see ENODEV specified as an errno that gets set by read which makes me think that this conditional can never be hit.",
      "range": {
        "startLine": 211,
        "startChar": 4,
        "endLine": 211,
        "endChar": 29
      },
      "revId": "0588ee5a0357c3372903f06abcdf181a0404d20d",
      "serverId": "adbd0a64-1f21-4d83-b585-671fe73cb6e4",
      "unresolved": true
    },
    {
      "key": {
        "uuid": "83151917_524f73dc",
        "filename": "src/PSUSensor.cpp",
        "patchSetId": 5
      },
      "lineNbr": 211,
      "author": {
        "id": 1001162
      },
      "writtenOn": "2021-08-05T23:56:40Z",
      "side": 1,
      "message": "Yeah from https://man7.org/linux/man-pages/man2/read.2.html\n\n\u003e On error, -1 is returned, and errno is set to indicate the error.\n\nAlso from `journalctl` while testing:\n`psusensor[1872]: No such device at /sys/class/hwmon/hwmon18/temp2_input, closing fd`",
      "parentUuid": "96b867ac_3206d62e",
      "range": {
        "startLine": 211,
        "startChar": 4,
        "endLine": 211,
        "endChar": 29
      },
      "revId": "0588ee5a0357c3372903f06abcdf181a0404d20d",
      "serverId": "adbd0a64-1f21-4d83-b585-671fe73cb6e4",
      "unresolved": false
    }
  ]
}